core/store/orm/orm.go

```diff
@@ -899,16 +899,93 @@ func (orm *ORM) TxAttempts(offset, limit int) ([]models.TxAttempt, int, error) {
 	return attempts, count, err
 }
 
-func (orm *ORM) UnconfirmedTxAttempts() ([]models.TxAttempt, error) {
+// HighestPricedUnconfirmedTxAttempts returns the highest priced TxAttempt for each unconfirmed Tx.
+func (orm *ORM) HighestPricedUnconfirmedTxAttempts() ([]models.TxAttempt, error) {
```

Comment:
Could we break this function up into some smaller functions with descriptive names please?